### PR TITLE
Add stable diffusion hire.fix pipeline

### PR DESCRIPTION
**Pull request by 1649759610**
### PR types

New features

### PR changes

Models

### Description

**generate images with hires.fix function.**

```python
import paddle
from stable_diffusion_hires_fix import StableDiffusionHiresFixPipeline
from ppdiffusers import EulerAncestralDiscreteScheduler

pipe = StableDiffusionHiresFixPipeline.from_pretrained("stabilityai/stable-diffusion-2", paddle_dtype=paddle.float16)
pipe.scheduler = EulerAncestralDiscreteScheduler.from_config(pipe.scheduler.config)

generator = paddle.Generator().manual_seed(5232132133)
prompt = "1 real girl, long black hair, detailed face, light smile, chinese style, hanfu"
image = pipe(prompt, guidance_scale=7.5, height=768, width=768, generator=generator, num_inference_steps=40, hires_ratio=0.5, hr_resize_width=768, hr_resize_height=1024, enable_hr=True).images[0]

image.show()

```
<center></center>

**Note**
The image generated by ppdiffusers has a little difference from webui's due to the difference of generated sigmas. Here show a picture from webui below.

 <center></center>

### CODE REVIEW

File: ppdiffusers/examples/community/stable_diffusion_hires_fix.py [hunk 1]
```diff
@@ -0,0 +1,764 @@
...
+
+            noise = randn_tensor(latents.shape, dtype=latents.dtype, generator="initial_generator")
+            latents = self.scheduler.add_noise(latents, noise, init_timestep)
+
+            # 11. Prepare extra step kwargs. TODO: Logic should ideally just be moved out of the pipeline
+            extra_step_kwargs = self.prepare_extra_step_kwargs("initial_generator", eta)
+
+            # 12. denoising on hires.fix steps
+            num_warmup_steps = len(timesteps) - hr_steps * self.scheduler.order
+            print("Step2: Perform denosing on hires.fix stage.")
```
Comment (JunnYu) [inline ppdiffusers/examples/community/stable_diffusion_hires_fix.py]: print 坑你都不要吧

File: ppdiffusers/examples/community/stable_diffusion_hires_fix.py [hunk 2]
```diff
@@ -0,0 +1,764 @@
...
+
+        # 4. Prepare timesteps
+        hr_steps = int(num_inference_steps * hires_ratio)
+        sample_steps = num_inference_steps - hr_steps
+        self.scheduler.set_timesteps(sample_steps)
+        timesteps = self.scheduler.timesteps
+
+        # 5. Prepare latent variables
+        if generator is not None:
+            paddle.Generator().states_["initial_generator"] = copy.deepcopy(paddle.Generator().states_[generator])
```
Comment (JunnYu) [inline ppdiffusers/examples/community/stable_diffusion_hires_fix.py:629]: 设置一个叫 二阶段的generator，如果二阶段的generator为空，那么我们就用一阶段的generator（不为空的情况下）

File: ppdiffusers/examples/community/stable_diffusion_hires_fix.py (hunk 1 above)
Comment (1649759610) [reply ppdiffusers/examples/community/stable_diffusion_hires_fix.py]: deleted

File: ppdiffusers/examples/community/stable_diffusion_hires_fix.py (hunk 2 above)
Comment (1649759610) [reply ppdiffusers/examples/community/stable_diffusion_hires_fix.py:629]: Better not to add the second generator, some reasons as follows:
- This usage may confused user with two random seed, and it seems unreasonable to control a image generation with two seeds.
- It maybe  result in  bad image generation.

**Review by JunnYu — APPROVED** [review]
APPROVE

**Pull request merged**